### ISSUE TITLE
Shorthand for combining losses at various weights

### ISSUE THREAD

**Issue by tomaarsen** [UKPLab/sentence-transformers#2440]
Hello!

## Context
This issue describes a feature that I am planning to be included in a release before v3, or alternatively, in v3 of Sentence Transformers.

## Details
When training embedding models, it is possible to want to include multiple losses for various datasets, e.g. see our [multi-task learning example](https://github.com/UKPLab/sentence-transformers/blob/0116721cc9862bfcc63ce1e736d7e9a706a645bc/examples/training/other/training_multi-task.py#L104-L107). Here we use SoftmaxLoss for NLI data and CosineSimilarityLoss for STSbenchmark data. However, it may also be interesting to use multiple loss functions for the same data. In this case, you can create your own loss function that internally calls the other loss functions & applies some weights to the losses, but it may be beneficial to create a weighted shorthand for this.

This does get somewhat tricky as different loss functions may require different inputs (positive pairs, text-positive-negative triplets, sentence_A-sentence_B-score triplets), etc.

This requires some more additional thought & experimentation. I haven't done enough multi-task testing myself to see if a weighted combined loss would be sufficiently valuable.

cc @bwanglzu @ir2718 @johneckberg @aamir-s18 as I know you're interested in my TODO list. 

- Tom Aarsen
